diff -upNr xapian-core-1.0.2.orign/xapian-config.in xapian-core-1.0.2/xapian-config.in
--- xapian-core-1.0.2.orign/xapian-config.in	2007-07-05 02:39:51.000000000 +0200
+++ xapian-core-1.0.2/xapian-config.in	2007-10-25 23:10:59.000000000 +0200
@@ -68,7 +68,6 @@ set_F_to_extra_ldflags() {
 # -L option required (if any).
 set_L_to_library_path() {
     L=
-    [ /usr/lib != "@libdir@" ] && L="-L@libdir@ "
 }
 
 # -I options for compiling against an uninstalled xapian.
